### DISCUSSION TITLE
mdraid resync speed after ignition boot (first boot)

### DISCUSSION THREAD

**Discussion by schmitch** [flatcar/Flatcar#616] [Q&A] [answered]
hello, we are building a nvme raid on top of mdraid with flatcar linux and we found out that it takes a really really long time to resync without raising `dev.raid.speed_limit_max`.

our ignition config is build like: https://www.flatcar.org/docs/latest/setup/storage/raid/
when we manually build the raid:

`mdadm --create /dev/md/data_array --level raid10 --name data_array --raid-devices=4 /dev/sda1 /dev/sdb1 /dev/sdc1 /dev/sdd1`

and call:

`sysctl -w  dev.raid.speed_limit_max=10000000`

it takes like 10 minutes to build the raid, however without it, it takes hours or even more and it's the first boot, is there any way to also specify the `dev.raid.speed_limit_max` when first boot kicks in?

**Comment by pothos** [comment, marked as answer]
Did you try setting it through a `/etc/sysctl.d/` entry? Otherwise https://www.flatcar.org/docs/latest/setup/customization/other-settings/#tuning-sysctl-parameters has some instructions, too

**Reply by schmitch** [reply]
will `files` be applied before `filesystem`?

**Reply by pothos** [reply]
afterwards

**Reply by schmitch** [reply]
I have no idea, but this does not look to have any effect, on first boot.

**Reply by schmitch** [reply]
(in fact it does not work)
so basically I either wait a day or configure mdraid manually or after first boot

**Reply by pothos** [reply]
So, `/etc/sysctl.d/` does not take effect (probably related to the module load time and the linked docs could help) but anyway you would want to assemble the raid in Ignition which is even earlier. The workaround I recommend is not doing in manually but having a systemd unit that does the call to `mdadm` and sets the `sysctl` prior to it.

**Reply by schmitch** [reply]
yeah that's what I figured, still thanks for your input
